base_layer/core/tests/block_validation.rs

```diff
@@ -682,32 +691,63 @@ async fn test_block_sync_body_validator() {
     let (tx04, _) = spend_utxos(
         txn_schema!(from: vec![outputs[3].clone()], to: vec![50_000 * uT], fee: 20*uT, lock: 2, features: OutputFeatures::default()),
     );
-    let (template, _) = chain_block_with_new_coinbase(&genesis, vec![tx01.clone(), tx02.clone()], &rules, &factories);
+
+    // Coinbase extra field is too large
+    let extra = iter::repeat(1u8).take(65).collect();
+    let (template, _) = chain_block_with_new_coinbase(
+        &genesis,
+        vec![tx01.clone(), tx02.clone()],
+        &rules,
+        &factories,
+        Some(extra),
+    );
+    let new_block = db.prepare_new_block(template).unwrap();
+    let max_len = rules.consensus_constants(0).coinbase_output_features_extra_max_length();
+    let err = validator.validate_body(new_block).await.unwrap_err(); // Whoops! This should not pass validation
+    assert!(
+        matches!(
+            err,
+            ValidationError::TransactionError(TransactionError::InvalidOutputFeaturesCoinbaseExtraSize{len, max }) if
+            len == 65 && max == max_len
+        ),
+        "{}",
+        err
+    );
```

Comment:
So this currently passes as a valid block. Unless I'm mistaken, this is a full blockchain test, and so all validation rules should be checked.